packages/realm-react/src/index.tsx

```diff
@@ -82,6 +80,44 @@ type RealmContext = {
    * @param deps - An array of dependencies that will be passed to {@link React.useMemo}
    * @returns a collection of realm objects or an empty array
    */
+
+  /**
+   * @overload
```

Comment:
In lieu of difficulties with the "overload" flag, why not just simplify the documentation down to what the desired usage is?
They types can still handle the previous usage.  My 2 cents.

Reply:
> the desired usage is?

Sure - but I don't know if I actually know what that is 🤔 I guess we could focus on passing an object and use this as a fix for #6259 and eventually make this more convenient via #6284?

Reply:
In my opinion, passing as an object is the desired usage.  If I would have thought of it earlier, it would be the only way to do it.

Reply:
If we do that, we could also `@deprecate` the existing signature for `useQuery`?

Reply:
Yes, I think that's the way forward.  We are still on version 0.x.y, so we can make some broader changes.  I think in the end it makes things more extensible if we want to add more options to the hooks

Reply:
I've updated the docs to focus solely on the signature where an options object is passed 👍

Reply:
The README should also be updated.  Or perhaps we should just simplify the README to point to the docs page 🤔

Reply:
I'm contemplating adding an overload, passing just the type as first argument and avoid deprecation of that:

```ts
const cars = useQuery(Car);
```

feels cleaner than

```ts
const cars = useQuery({ type: Car });
```

Reply:
Yeah, that's cleaner, but it's not "that" bad.